python/lsst/ts/salobj/sal_info.py

```diff
@@ -1207,126 +1226,137 @@ async def _read_loop(self) -> None:
                         self.start_task.set_result(None)
                     continue
-                message_error = message.error()
-                if message_error is not None:
-                    sequential_read_errors += 1
-                    self.log.warning(
-                        f"Ignoring Kafka message with error {message_error!r}"
-                    )
-                    if sequential_read_errors > MAX_SEQUENTIAL_READ_ERRORS:
-                        raise RuntimeError("Too many sequential read errors; giving up")
-                    continue
-
-                kafka_name = message.topic()
-                if kafka_name is None:
-                    sequential_read_errors += 1
-                    self.log.warning("Ignoring Kafka message with null topic name")
-                    if sequential_read_errors > MAX_SEQUENTIAL_READ_ERRORS:
-                        raise RuntimeError("Too many sequential read errors; giving up")
-                    continue
-
-                sequential_read_errors = 0
-
-                read_topic = self._read_topics[kafka_name]
-
-                deserializer, context = self._deserializers_and_contexts[kafka_name]
-                try:
-                    data_dict = deserializer(message.value(), context)
-                except (SchemaResolutionError, SerializationError):
-                    if kafka_name not in schema_resolution_errors:
-                        self.log.exception(
-                            f"Failed to deserialize {kafka_name}. This usually means the topic was published "
-                            "with an incompatible version of the schema from this instance. You might need "
-                            "to check which component has the incompatible schema and update it. It could be "
-                            "that the publisher is incompatible or this instance reading the topic. "
-                            "If this is a single old historical data, it might be safe to ignore this "
-                            "error. Additional deserialization error messages of this topic will be "
-                            "suppressed and will show as shorter error messages every "
-                            f"{SCHEMA_RESOLUTION_LOG_ERROR_THRESHOLD} failed attempt. If this error persist "
-                            "you should investigate it further."
-                        )
-
-                        schema_resolution_errors[kafka_name] = 0
-                    elif (
-                        schema_resolution_errors[kafka_name]
-                        % SCHEMA_RESOLUTION_LOG_ERROR_THRESHOLD
-                        == 0
-                    ):
-                        self.log.error(
-                            f"Failed to deserialize {schema_resolution_errors[kafka_name]} samples of "
-                            f"{kafka_name}. Check schema compatibility!"
+                for message in messages:
+                    message_error = message.error()
```

Comment:
Keeping message processing leads to quite big indentation, which makes code harder to read. Can you please create a method to process messages, so the code (assuming it's called _process_message) will shrink to:
```
for message in messages:
    self._process_messaeg(message)
```